packages/angular_devkit/benchmark/src/monitored-process_spec.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { dirname } from 'path';
+import { tap } from 'rxjs/operators';
+import { Command } from './command';
+import { LocalMonitoredProcess } from './monitored-process';
+
+describe('LocalMonitoredProcess', () => {
+  const cmd = new Command(
+    'node',
+    ['test-script.js'],
+    dirname(require.resolve('./test/test-script.js')),
+  );
+
+  it('works', (done) => {
+    const process = new LocalMonitoredProcess(cmd);
+    process.run().pipe(
+      tap(code => expect(code).toBe(0)),
+    ).toPromise().then(done, done.fail);
+  });
+
+  it('captures stdout', (done) => {
+    const process = new LocalMonitoredProcess(cmd);
+    const stdoutOutput: string[] = [];
+    process.stdout$.subscribe(data => stdoutOutput.push(data.toString()));
+
+    process.run().pipe(
+      tap(() => expect(stdoutOutput).toEqual(['stdout start\n', 'stdout end\n'])),
+    ).toPromise().then(done, done.fail);
+  });
+
+  it('captures stderr', (done) => {
+    const process = new LocalMonitoredProcess(cmd);
+    const stdoutOutput: string[] = [];
+    process.stderr$.subscribe(data => stdoutOutput.push(data.toString()));
+
+    process.run().pipe(
+      tap(() => expect(stdoutOutput).toEqual(['stderr start\n', 'stderr end\n'])),
+    ).toPromise().then(done, done.fail);
+  });
+
+  it('captures stats', (done) => {
+    const process = new LocalMonitoredProcess(cmd);
+    const statsOutput: string[] = [];
+    process.stderr$.subscribe(data => statsOutput.push(data.toString()));
+
+    process.run().pipe(
+      tap(() => expect(statsOutput.length).toBeGreaterThan(0)),
+    ).toPromise().then(done, done.fail);
```

Comment:
curious, would it be more readable with `async`/`await` ?

Reply:
It goes from
```
  it('works', (done) => {
    const process = new LocalMonitoredProcess(cmd);
    process.run().pipe(
      tap(code => expect(code).toBe(0)),
    ).toPromise().then(done, done.fail);
  });
```

to

```
  it('works', async () => {
    const process = new LocalMonitoredProcess(cmd);
    const res = await process.run().pipe(toArray()).toPromise();
    expect(res).toEqual([0]);
  });
```

I think it looks nicer and also ensures the number of emissions.